sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/ViewRegistry.java

```diff
@@ -36,39 +54,70 @@ Batcher createBatcher(
       MeterSharedState meterSharedState,
       InstrumentDescriptor descriptor) {
 
-    Aggregation aggregation = getRegisteredAggregation(descriptor);
+    AggregationConfiguration specification = findBestMatch(descriptor);
 
-    // todo: don't just use the defaults!
-    switch (descriptor.getType()) {
-      case COUNTER:
-      case UP_DOWN_COUNTER:
-      case SUM_OBSERVER:
-      case UP_DOWN_SUM_OBSERVER:
-        return Batchers.getCumulativeAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
-      case VALUE_RECORDER:
-        // TODO: Revisit the batcher used here for value observers,
-        // currently this does not remove duplicate records in the same cycle.
-      case VALUE_OBSERVER:
-        return Batchers.getDeltaAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
+    Aggregation aggregation = specification.aggregation();
+
+    if (Temporality.CUMULATIVE == specification.temporality()) {
+      return Batchers.getCumulativeAllLabels(
+          descriptor, meterProviderSharedState, meterSharedState, aggregation);
+    } else if (Temporality.DELTA == specification.temporality()) {
+      return Batchers.getDeltaAllLabels(
+          descriptor, meterProviderSharedState, meterSharedState, aggregation);
     }
-    throw new IllegalArgumentException("Unknown descriptor type: " + descriptor.getType());
+    throw new IllegalStateException("unsupported Temporality: " + specification.temporality());
   }
 
-  private static Aggregation getRegisteredAggregation(InstrumentDescriptor descriptor) {
-    // todo look up based on fields of the descriptor.
+  // todo: consider moving this method to its own class, for more targeted testing.
+  private AggregationConfiguration findBestMatch(InstrumentDescriptor descriptor) {
```

Comment:
Mostly makes sense but I'm wondering why is this a best match? Shouldn't the aggregation configuration be more precise?

Reply:
Well, I can imagine someone saying 
1) By default, aggregate all ValueRecorders with MinMaxSumCount.
2) This particular ValueRecorder, named "http.request.latency"...that one I want Histograms for

So, "best match" will find the most precise selection and choose that, falling back to less precise ones. Is there a better name I could use?

Reply:
I'd just name the method `getAggregation` and the loop can have a comment `// See if user defined a specific configuration for this descriptor`, the definition itself seems precise, not best-effort.